packages/data/src/test/registry.js

```diff
@@ -416,10 +416,6 @@ describe( 'createRegistry', () => {
 			shouldThrow = true;
 
 			registry.select( 'demo' ).getValue();
-
-			process.nextTick( () => {
-				done();
-			} );
```

Comment:
So is `done` not passed as an argument anymore? Given that this test was testing some behavior that would expect to resolve on the next tick, I'd be worried about changing this to be synchronous. It would be good to verify the failure case (by revising the implementation) is still caught.

If my understanding of how they're promoting to return promises, I think we could still refactor this?

```js
return new Promise( ( resolve ) => process.nextTick( resolve ) );
```

?

Or if we're already relying on `Promise.resolve` in the `resolvers` implementation above, I'd be a bit curious if it would be enough to:

```js
return Promise.resolve();
```

(If it does work, a comment clarifying its purpose would be a good idea)

Reply:
Tests pass 🤷‍♂ Anyway, I can give it a try to use one of the suggested approaches.

Reply:
I think the tests are relying on the async behavior in order to verify a failure, so it's not surprising to me that they'd pass if it's changed to be synchronous. Perhaps the test can be improved to be less fragile if it currently relies on this. This is part of what I was hoping by "verify the failure case", though it may not be straight-forward to pinpoint the exact logic responsible for making sure this is working correctly.

Reply:
Updated with `return new Promise( ( resolve ) => process.nextTick( resolve ) );`.